python/dcr_template.py

```diff
@@ -890,16 +959,42 @@ def build_model(self, use_full=True, use_regularization=True, use_only_detected=
                                     of detected sources.
         @param verbose  Flag, set to True to print progress messages.
         """
-        dcr_kernel = self.build_dcr_kernel(use_full=use_full)
-        self.model = np.zeros((self.n_step, self.y_size, self.x_size))
-        self.weights = np.zeros_like(self.model)
+        kernel_base = self.build_dcr_kernel(self.exposures, use_full=False, use_psf=False)
+        kernel_weight = divide_kernels(self.build_dcr_kernel(self.exposures, use_full=True, use_psf=True),
+                                       self.build_dcr_kernel(self.exposures, use_full=False, use_psf=True))
+        if use_regularization:
+            if kernel_weight is None:
+                regularize_scale = np.max(kernel_base)
+            else:
+                regularize_scale = np.sqrt(np.max(kernel_base)*np.max(kernel_weight))
+            if positive_regularization:
+                test_values = np.ones(self.n_images*self.kernel_size**2)
+                test_solution = self.solve_model(self.kernel_size, self.n_step, kernel_base, test_values,
+                                                 use_nonnegative=use_nonnegative, use_regularization=False,
+                                                 kernel_weight=kernel_weight)
+            else:
+                test_solution = None
+            self.regularize = self.build_regularization(x_size=self.kernel_size, y_size=self.kernel_size,
+                                                        n_step=self.n_step, weight=regularize_scale,
+                                                        frequency_regularization=frequency_regularization,
+                                                        positive_regularization=positive_regularization,
+                                                        test_solution=test_solution)
+        else:
+            self.regularize = None
+        lstsq_kernel = self.build_lstsq_kernel(kernel_base, regularization=self.regularize,
+                                               use_regularization=use_regularization,
+                                               kernel_weight=kernel_weight)
+        model = np.zeros((self.n_step, self.y_size, self.x_size))
+        weights = np.zeros((self.y_size, self.x_size))
         pix_radius = self.kernel_size//2
 
         variance = np.zeros((self.y_size, self.x_size))
         for exp in self.exposures:
             variance += exp.getMaskedImage().getVariance().getArray()**2.
         self.variance = np.sqrt(variance) / self.n_images
         detected_bit = exp.getMaskedImage().getMask().getPlaneBitMask("DETECTED")
+        image_arr = [exp.getMaskedImage().getImage().getArray() for exp in self.exposures]
```

Comment:
I spent a while thinking about this, and finally did come up with an alternative that reduces the number of loops by one (and runs much faster). The results are very slightly different, but I think I can establish that they are at least as accurate.